x/revenue/v1/keeper/evm_hooks.go

```diff
@@ -81,8 +81,12 @@ func (k Keeper) PostTxProcessing(
 
 	// get available precompiles from evm params and check if contract is in the list
 	if containsPrecompile {
-		if err := k.distributionKeeper.FundCommunityPool(ctx, fees, k.accountKeeper.GetModuleAddress(k.feeCollectorName)); err != nil {
-			return err
+		// when baseFee and minGasPrice in freemarker module are both 0
+		// the user may send a transaction with gasPrice of 0 to the precompiled contract
+		if developerFee.IsPositive() {
```

Comment:
this should return `nil` on line 46

```
if contract == nil || msg.GasPrice().Sign() == 0 {
   return nil
}
```

Reply:
There is no early **return nil** for this situation because I consider that events will be issued later, even though the developerFee is 0aevmos. But I think your suggestion is more reasonable. Modifications have been made as per your suggestion.